src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,113 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    static final int DEFAULT_INITIAL_CAPACITY = 1 << 4;
+    static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private Node<K, V>[] table;
+    private int size;
+    private int threshold = (int)(DEFAULT_INITIAL_CAPACITY * DEFAULT_LOAD_FACTOR);
+
+    private class Node<K, V> {
+        private final int hash;
+        private final K key;
+        private V value;
+        private Node<K, V> next;
+
+        public Node(int hash, K key, V value, Node<K, V> next) {
+            this.hash = hash;
+            this.key = key;
+            this.value = value;
+            this.next = next;
+        }
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (table == null) {
+            table = new Node[DEFAULT_INITIAL_CAPACITY];
+            table[0] = new Node<>(hash(key), key, value, null);
+            size++;
+        } else if (containsKey(key)) {
+            Node<K,V> currentNode = getNode(key);
+            currentNode.value = value;
+        } else {
+            int index = hash(key) % table.length;
+            if (table[index] == null) {
+                table[index] = new Node<>(hash(key), key, value, null);
+            } else {
+                Node<K, V> current = table[index];
+                while (current.next != null) {
+                    current = current.next;
+                }
+                current.next = new Node<>(hash(key), key, value, null);
+            }
+            size++;
+        }
+        if (size >= threshold) {
+            resize();
+        }
     }
 
     @Override
     public V getValue(K key) {
-        return null;
+        Node<K, V> currentNode = getNode(key);
+        return currentNode == null ? null : currentNode.value;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    public boolean containsKey(K key) {
+        return getNode(key) != null;
+    }
+
+    private int hash(Object key) {
```

Comment:
This is not the best way to override the hash code. I advise you to look at how it is implemented in IDEA